fem/qfunction.cpp

```diff
@@ -31,6 +31,7 @@ QuadratureFunction &QuadratureFunction::operator=(const Vector &v)
 
 
 QuadratureFunction::QuadratureFunction(Mesh *mesh, std::istream &in)
```

Comment:
So, I think this will cause issues as `Vector::Load` does not make sure that the data first lives on the host. However, we assume here that the data should now be on the device when the `Load` call is made. It seems like we should fix `Vector::Load` so it can't fail as it looks like this might also be a potential issue with how a couple of the `GridFunction` constructors work as well:
https://github.com/mfem/mfem/blob/master/fem/gridfunc.cpp#L36 . 

Now that I'm looking at some of those constructors it does appear that there could be some potential bugs when running on the GPU as its doing host-memory operations even though it's asked the data to be on the device beforehand. However, I'm a bit fuzzy on some of the memory manager aspects, so @v-dobrev or @camierjs could maybe correct me here.

Reply:
Yeah, maybe `Vector::Load` should have `HostWrite` at the beginning?

Reply:
One has to be careful with the method `Vector::UseDevice(bool)` -- it may not be exactly what you expect -- here's the Doxygen: https://github.com/mfem/mfem/blob/ea5c4664864cb84c0a453f527d6befec05a8b8a9/linalg/vector.hpp#L108-L117

Reply:
Right, that's useful documentation.

I added `HostWrite` to `Vector::Load`. This will prevent a failure if the memory has flag `VALID_DEVICE` when `Load` is called.